### DISCUSSION TITLE
Negative values in `Community.knockout_taxa()` using  “relative change” method

### DISCUSSION THREAD

**Discussion by alejandrocs98** [micom-dev/micom#154] [Q&A] [answered]
Hi!
I have a couple of questions about the `Community.knockout_taxa()` method.
1) I followed the video of one of the editions of the ISB courses (https://www.youtube.com/watch?v=GfaglaE1br0) and performed the taxa knockout analysis using "relative change" as method (like in the video). In the video, they mention that positive values indicate competition and negative values indicate trophic interactions. As it is a difference it sounds reasonable to me, but I checked the source code and when using this method the results one obtains are: `new /= old`, which would suggest that values over 1 indicate competition and lower than 1 indicate trophic interactions. The problem is that in the video and actually in my results it is possible to get negative values for this analysis, which should imply that one of the calculated growth rates is negative. Is that correct? What is the interpretation for that?
2) My second question is more straightforward, I understand that in `Community.knockout_taxa()` method the growth rates are obtained via cooperative tradeoff method. For this it uses **pFBA** or **minimal imports** approach? And in the case I use "raw" as a method, could I compare the values obtained with the growth rates I got by using the `grow` workflow?
Thank you in advance!

**Comment by cdiener** [comment, marked as answer]
Hi, sure, let's get to it :)

1. That depends on what you use as the strategy. In the course `relative change` was used. If you look at the [code](https://github.com/micom-dev/micom/blob/2f5b425c2816aee90d5623b7d46f9a78c5c7fecf/micom/problems.py#L132) you see that it looks for the words `change` and then `relative`. So both if statements would trigger meaning it first calculates `new = new - old` followed by `new /= old`. You are right thought that you could hack it by just passing the strategy `relative` which is a bug :bug:

2. Neither, and it would not matter. Cooperative tradeoff is enough to fix the growth rates because it will give a unique solution to each taxon. For minimal imports or pFBA the growth rates would be fixed to those values, so they wouldn't change. `knockout_taxa` only looks at the growth rates. And yes they would be comparable to what you get from the grow workflow as long as you use the same medium and tradeoff value.

Hope that helps,
Best

**Comment by alejandrocs98** [comment]
Thank you!! :)
